src/ocean/core/Option.d

```diff
@@ -0,0 +1,236 @@
+
+    Wraps any type in a struct that also contains boolean field indicating if
+    value is in defined state.
+
+    If T is a value type, `Option!(T)` is value type too.
+
+    Copyright:
+        Copyright (c) 2009-2016 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+
+*******************************************************************************/
+
+module ocean.core.Option;
+
+version(UnitTest)
+{
+    import ocean.core.Test;
+}
+
+/// ditto
+struct Option ( T )
+{
+    /// Alias to make code working with undefined state more readable
+    public const undefined = Option!(T).init;
+
+    /// wrapped arbitrary value
+    private T value;
+
+    /// flag indicating if stored value is in defined state
+    private bool defined = false;
+
+    /**************************************************************************
+
+        Puts `this` into defined state
+
+        Params:
+            rhs = value to assign
+
+    **************************************************************************/
+
+    public void opAssign ( T rhs )
+    {
+        this.defined = true;
+        this.value = rhs;
+    }
+
+    /**************************************************************************
+
+        Puts `this` into undefined state
+
+    **************************************************************************/
+
+    public void reset()
+    {
+        this.defined = false;
+    }
+
+    /**************************************************************************
+
+        Interface to retrieve stored value. It is intentionally designed in a
+        way that forces you to handle "undefined" state to avoid issues akin
+        to "null pointer".
+
+        This will become more convenient with D2 when lambda type inference
+        will happen, as well as shorter notation:
+
+        ```
+            value.visit(
+                ()  => { },
+                (x) => { }
+            );
+        ```
+
+        Parameters:
+            cb_undefined = action to take if value is not defined
+            cb_defined   = ditto for defined. May modify internal value via
+                reference
+
+    **************************************************************************/
+
+    public void visit ( void delegate() cb_undefined,
+        void delegate(ref T) cb_defined )
+    {
+        if (this.defined)
+            cb_defined(this.value);
+        else
+            cb_undefined();
+    }
+
+    /**************************************************************************
+
+        A more "old-school" version of `visit`. Provided both to conform style
+        of existing code and avoid delegate bugs in dmd1.
+
+        Discouraged by default as more error-prone than `visit`.
```

Comment:
ok, this might be a reason why `get` is not inside release notes.

Reply:
Correct. I added docs and tests for it though.